### ISSUE TITLE
Post editor autoformatting is severely broken

### ISSUE THREAD

**Issue by gwillen** [Discordius/Lesswrong2#314]
I don't know of a nicer or more precise way to say this because I haven't really figured out the pattern. But try typing or pasting a string containing several underscores, then go somewhere else in the document and press the spacebar: the editor will do something bizzare and nonsensical reflecting probably multiple different bugs in how it autoformats underscores into italics, and how it handles computing indices/ranges into the string being edited. It's really, really bad. :-(

This is especially a problem because I'm trying to make a post containing links with underscores, and it's just not possible as far as I can tell?

**Comment by gwillen** [comment]
As an example, if I type the string:

"hello_world_one_two_three_four_five_six_seven_eight"

And then press the spacebar, it is rendered into:

"helloworld one_two hree_ffour ve_sisix en_eight" 

And the cursor is left before "en", and "world", "two", "four", and "six" are italicized.

**Comment by gwillen** [comment]
FWIW it seems somewhat misguided to me to do this at all -- why is a WYSIWYG editor turning underscores into italics when I can just click italicize? Wouldn't it be more useful to allow people to use underscores in their posts?

**Comment by Discordius** [comment]
Yeah, I am conflicted about this. People really like markdown, and a bunch of people want to be able to use it, but I also want to stay in the WYSIWYG domain, and so we have the current ugly chimera. We might end up removing markdown support from the editor if we find that not that many people actually use it, but I am unsure.

**Comment by FerociousTruth** [comment]
This is a situation where modes are appropriate.  There is no way to get a good experience mixing WYSIWYG with markdown and LaTex.  For example, I considered crossposting my blog post.  However, when I pasted in the markdown, the markdown wasn't correctly processed.  I later figured out that I could choose "Paste as Plain Text" and that would make it work better, but it still got lots of things wrong. (I've just done a link post instead.)

I suggest either giving the user a setting about which editor they prefer or just remembering which editor they last used (perhaps with a separate value for the comment editor from the post editor).

**Comment by gwillen** [comment]
I hate to harp on this, but since people are talking about promoting lesserwrong out of beta: I don't understand how "it's not possible to use underscores in comments" is an acceptable state of the editor. I hope I'm missing something, but if I'm not, can the automatic markdown interpretation please get turned off? :-\
